src/commandSocket.h

```diff
@@ -7,95 +7,62 @@
 #include "Progress.h"
 #include "PrintFeature.h"
 
-#include <memory>
-
-#ifdef ARCUS
-#include "Cura.pb.h"
-#endif
-
 namespace cura
 {
 
 class CommandSocket
 {
 private:
-    static CommandSocket* instance; //!< May be a nullptr in case it hasn't been instantiated.
-
-    CommandSocket(); //!< The single constructor is known only privately, since this class is similar to a singleton class (except the single object doesn't need to be instantiated)
+    static std::unique_ptr<CommandSocket> instance_; //!< always instantiated
```

Comment:
Why does instance_ have an underscore at the end?
There is no code convention to that end.